### ISSUE TITLE
[BEP] Build output log not updating 

### ISSUE THREAD

**Issue by jerrymarino** [bazelbuild/tulsi#58]
Hello Tulsi team! I'm currently using `TulsiGenerator.framework` and `bazel_build.py` script in the context of an Xcode project generator for Bazel, [XCHammer](https://github.com/pinterest/xchammer). Overall, the setup is working quite well, so thanks to you all 🎉 !

## Issue context

I'm in the process of [updating to Bazel `0.16.1`](https://github.com/pinterest/xchammer/pull/34) and the HEAD of Tulsi. After the update, I noticed the build log isn't updated until the build is complete. There is no way to see the `progress` of the build as it executes actions.

## What I've tried so far

After some testing, I found that the BEP json file is not updated until the build completes. I was wondering if anyone had any tips on what to do. Is there a hidden bit I can flip to stream all progress?

It looks like setting `--build_event_publish_all_actions=true` will cause Bazel to update the BEP json file more frequently. Still, it isn't nearly alive as the standard output UI.

**Comment by sergiocampama** [comment]
check out this line https://github.com/bazelbuild/tulsi/blob/master/src/TulsiGenerator/Scripts/bazel_build.py#L217

**Comment by jerrymarino** [comment]
Hey @sergiocampama - I'm currently setting `bes_outerr_buffer_size=0`

Here's a set of Bazel settings I'm reproducing this with locally:

```
--verbose_failures
--bes_outerr_buffer_size=0
--apple_platform_type=ios
--cpu=ios_x86_64
--watchos_cpus=i386
--xcode_version=9.3
--tool_tag=tulsi_vUNKNOWN:bazel_build
--build_event_json_file=/tmp/t.json
--noexperimental_build_event_json_file_path_conversion
--aspects
@tulsi//tulsi:tulsi_aspects.bzl%tulsi_outputs_aspect
--output_groups=tulsi_outputs,default
//ios-app:ios-app
--override_repository=tulsi=/Users/jerry/Projects/xchammer-github/.build/debug/tulsi-aspects
```

Under a 30 second build with 150 actions, the file is only written 2-3 times total. Is that expected/ on point with what you're seeing? Looking at just [SynchronizedOutputStream](https://github.com/bazelbuild/bazel/blob/ad0d7e946cd2ad05ec89c78a2f1959ffbbfa12a8/src/main/java/com/google/devtools/build/lib/runtime/SynchronizedOutputStream.java), it seems like it _should_ be more frequent.

**Comment by jerrymarino** [comment]
@DavidGoldman thank you kindly and glad that this is sorted out! 👍

**Comment by DavidGoldman** [comment]
This should be fixed if you use [Bazel 0.17.1](https://github.com/bazelbuild/bazel/releases/tag/0.17.1) or later.
